source/material_model/rheology/friction_models.cc

```diff
@@ -359,6 +364,7 @@ namespace aspect
             && in.current_cell.state() == IteratorState::valid)
           {
             const double fault_volume = get_fault_volume(volume_fractions);
+            // ToDo: make an input parameter, how much fault there needs to be?
```

Comment:
Actually, I'm wondering whether instead of checking whether the fault_volume is larger than 0.5, you should just only fill the reaction term for the RSF fields that could have a nonzero theta (`RSF_composition_masks`), like you do below. So remove the fault_volume>0.5 completely. Would that give problems outside of the fault because theta could/would be 0 there? Then there could be a check that if theta is zero, then the reaction term is 0, or something.

Reply:
hm, I think I cant really follow your thoughts here. 
The `fault_volume `is based on the `RSF_composition_masks` by adding up the volume fractions of all RSF_compositions. So I do only fill the reaction terms for theta where it could be non-zero.

Reply:
Ah I see, but that means you don't fill the reaction terms for where theta is non-zero, but for where the fault volume is larger than 0.5. They could be slightly different things right? Because there is still some fault material even if 0>fault_volume<0.5. Therefore I was wondering if you can compute the reaction terms for all the RSF_composition_masks, regardless of whether their combined fault_volume>0.5 or not. Outside the fault, updates to theta should be small because of the strain rate/velocity being small right?